test/library/packages/Sort/RadixSort/radixMSBModule.chpl

```diff
@@ -0,0 +1,83 @@
+ use BitOps;
+ use Sort;
+ use Random;
+ config param bucketBits=8; 
+
+ module MSBmodule {
+
+   proc radixSortMSB(array:[?dom] int) {
+     var auxArray:[1..array.size] int;
+     var place = findPlace(array);
+     __radixSortMSB(array,place,dom.low,dom.high,auxArray);
+   }
+  
+   private proc __radixSortMSB(array:[] int, place: int, startIndex: int, endIndex: int, auxArray:[] int) {
+
+     param numBuckets = (1 << bucketBits);
+     var counts:[1..numBuckets+1] int;  
+     var currentDigit: int; 
+     var countIndex: int;
+     var auxIndex: int;
+   
+     //base case, exit condition
+     if (endIndex<=startIndex || place<0) {
+       return;
+     }
+   
+     for i in startIndex..endIndex {
+       // find the postion to store a particular number in the counts array
+       currentDigit = findPosition(place, array[i], numBuckets);
+     
+       //increment count at that position
+       counts[currentDigit+1] += 1; //increment count at the position of the m.s.d.
+     }
+
+     //update counts array
+     counts = + scan counts;
+
+     for i in startIndex..endIndex {
+       countIndex = findPosition(place,array[i], numBuckets);
+       auxIndex = counts[countIndex+1];     
+       auxArray[auxIndex] = array[i];   
+       counts[countIndex+1] -=1;                 
```

Comment:
This still writes elements to auxArray "backwards" since counts[countIndex+1] is subtracted from here. Can we make curOffsets = 0, count[0], count[0] + count[1], ... and then add 1 to curOffsets?